contrib/completion/_fd

```diff
@@ -0,0 +1,210 @@
+
+    + '(long-listing)' # long-listing output
+    '(abs-path null-sep max-results exec-cmds)'{-l,--list-details}'[use a long listing format with file metadata]'
+
+    + '(max-results)' # max number of results
+    '(long-listing exec-cmds)--max-results=[limit number of search results to given count and quit]:count'
+    '(long-listing exec-cmds)-1[limit to a single search result and quit]'
+
+    + '(error-messages)' # show FS errors
+    $no'--show-errors'
+
+    + '(file-system)' #
+    "--one-file-system[don't descend into directories on other file systems]"
+    '!--mount'
+    '!--xdev'
+
+    + dir-depth # directory depth
+    '(--exact-depth -d --max-depth)'{-d+,--max-depth=}'[set max directory depth to descend when searching]:depth'
+    '!(--exact-depth -d --max-depth)--maxdepth:depth'
+    '(--exact-depth --min-depth)--min-depth=[set directory depth to descend before start searching]:depth'
+    '(--exact-depth -d --max-depth --maxdepth --min-depth)--exact-depth=[only search at the exact given directory depth]:depth'
+
+    + filter-misc # Filter search
+    '*'{-t+,--type=}"[filter search by type]:type:(($fd_file_types))"
+    '*'{-e+,--extension=}'[filter search by file extension]:extension'
+    '*'{-E+,--exclude=}'[exclude files/directories that match the given glob pattern]:pattern'
+    '*'{-S+,--size=}'[limit search by file size]:size limit'
+    '(-o --owner)'{-o+,--owner=}'[filter by owning user and/or group]:owner/group'
+
+    + ignore-file
+    '*--ignore-file=[add a custom, low-precedence ignore-file with .gitignore format]: :_files'
+
+    + '(filter-mtime-newer)' # Filter files modified after than
+    '--changed-within=[limit search to files/directories modified within the given date/duration]:date/duration'
+    '!--change-newer-than=:date/duration'
+    '!--newer=:date/duration'
+
+    + '(filter-mtime-older)' # Filter files modified before than
+    '--changed-before=[limit search to files/directories modified before the given date/duration]:date/duration'
+    '!--change-older-than=:date/duration'
+    '!--older=:date/duration'
+
+    + '(color)' # all other options
+    {-c+,--color=}'[declare when to colorize search results]:when to colorize:((
+      auto\:"show colors if the output goes to an interactive console"
+      never\:"do not use colorized output"
+      always\:"always use colorized output"
+    ))'
+
+    + '(threads)'
+    {-j+,--threads=}'[set the number of threads for searching and executing]:threads'
+
+    + '(exec-cmds)' # execute command
+    '(long-listing max-results)'{-x+,--exec=}'[execute command for each search result]:command: _command_names -e:*\;::program arguments: _normal'
+    '(long-listing max-results)'{-X+,--exec-batch=}'[execute command for all search results at once]:command: _command_names -e:*\;::program arguments: _normal'
+
+    + other
+    $no'(--max-buffer-time)--max-buffer-time=:milliseconds'
```

Comment:
As with `--size`, you'd probably do like `buffer time (ms)`. If there is a pre-defined default, you could also put that in square brackets, as in `buffer time (ms) [1000]`

Reply:
I set it to `set amount of time to buffer before showing output (ms)`

Reply:
By convention, the units/default go in the description for the argument itself, not the option. See here for examples:

https://github.com/zsh-users/zsh/blob/master/Etc/completion-style-guide#L66

That way, when the user does like `--max-buffer-time <TAB>`, they can still get a complete explanation of what the expected value is (even if zsh doesn't actually have any possibilities to offer).

Reply:
I've updated to `[set amount of time to buffer before showing output]:time (ms)`. Now the unit is in the completion message for the argument.